### PR TITLE
Added new creature extra flag which makes npc visible only to dead units

### LINKED ISSUE #19333: [NPC][Naxx] Gothik encounter, some spectral NPCs are not visible.
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

**Description:**

So, two NPCs on Gothik encounter are not visible on "dead" side.

**Current behaviour:**

Two NPCs are not visible:
1) Spectral Death Knight, id = 16148
2) Spectral Rider, id = 16150
They are only visible on GM state.

**Expected behaviour:**

These NPCs must be visible after "transforming" into undead on "dead" side.

**Steps to reproduce the problem:**

1. Pull Gothik on any difficulty (with at least 1 player on each side)
2. Wait for around 5-6 wave.

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

TrinityCore rev. 5d562556c35a+ 2017-03-18 11:07:41 +0100 (3.3.5 branch) (Win64, RelWithDebInfo, Static) (worldserver-daemon)

**TDB version:**  335.62

**Operating system:** Win 8 x64 with all system updates.

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (For fixes containing C++, create a Pull Request.)

### PR DESCRIPTION
**Pull request by ghost**
**Target branch(es):** 3.3.5

- [x] 3.3.5

**Issues addressed:** closes #19333

**Tests performed:** Compiles

### CODE REVIEW

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -1016,7 +1016,7 @@ bool Creature::Create(ObjectGuid::LowType guidlow, Map* map, uint32 phaseMask, u
     LastUsedScriptID = GetCreatureTemplate()->ScriptID;
 
     /// @todo Replace with spell, handle from DB
```
Comment (Treeston) [inline src/server/game/Entities/Creature/Creature.cpp]: Remove todo?

**Review by Treeston — APPROVED** [review]
:+1:

**Comment by Keader** [comment]
Note: After Merge, update wiki please

File: src/server/game/Entities/Creature/Creature.cpp (hunk 1 above)
Comment (tkrokli) [inline src/server/game/Entities/Creature/Creature.cpp]: Can be done via web merge, if he doesn't want to add an extra commit.

**Pull request merged**